pkg/util/restoptions/hacks.go

```diff
@@ -0,0 +1,79 @@
+package restoptions
+
+import (
+	genericrest "k8s.io/kubernetes/pkg/registry/generic"
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/storage"
+	etcdstorage "k8s.io/kubernetes/pkg/storage/etcd"
+	"k8s.io/kubernetes/pkg/storage/storagebackend"
+	"k8s.io/kubernetes/pkg/storage/storagebackend/factory"
+)
+
+// Temporary hack until rebase of upstream PR 37770 ///////////////////////////////////////////////
```

Comment:
I'm not sure we'll want to cherry-pick this from 1.6 (it's pretty big) and the 1.5 rebase hasn't landed yet. Not sure about the desired time frame for this feature, but @mfojtik will know that.